* Rename Python2 related functions to Python generic ones [1]
* Add support for FreeBSD
* Add detection for PowerPC64

[1] Obtained from:

https://gitlab.archlinux.org/archlinux/packaging/packages/qt5-webengine/-/raw/main/qt5-webengine-python3.patch?ref_type=heads

--- mkspecs/features/functions.prf.orig	2023-07-20 09:52:10 UTC
+++ mkspecs/features/functions.prf
@@ -44,11 +44,11 @@ defineReplace(pythonPath) {
 
 # Returns the unquoted path to the python executable.
 defineReplace(pythonPath) {
-    isEmpty(QMAKE_PYTHON2) {
+    isEmpty(QMAKE_PYTHON) {
         # Fallback for building QtWebEngine with Qt < 5.8
-        QMAKE_PYTHON2 = python
+        QMAKE_PYTHON = python
     }
-    return($$QMAKE_PYTHON2)
+    return($$QMAKE_PYTHON)
 }
 
 # Returns the python executable for use with shell / make targets.
@@ -89,6 +89,10 @@ defineReplace(gnWebEngineArgs) {
         include($$QTWEBENGINE_ROOT/src/buildtools/config/windows.pri)
         include($$QTWEBENGINE_ROOT/src/core/config/windows.pri)
     }
+    freebsd {
+        include($$QTWEBENGINE_ROOT/src/buildtools/config/freebsd.pri)
+        include($$QTWEBENGINE_ROOT/src/core/config/linux.pri)
+    }
     isEmpty(gn_args): error(No gn_args found please make sure you have valid configuration.)
     return($$gn_args)
 }
@@ -98,6 +102,7 @@ defineReplace(gnPdfArgs) {
     macos: include($$QTWEBENGINE_ROOT/src/buildtools/config/mac_osx.pri)
     ios: include($$QTWEBENGINE_ROOT/src/pdf/config/ios.pri)
     win32: include($$QTWEBENGINE_ROOT/src/buildtools/config/windows.pri)
+    freebsd: include($$QTWEBENGINE_ROOT/src/buildtools/config/freebsd.pri)
     include($$QTWEBENGINE_ROOT/src/pdf/config/common.pri)
     isEmpty(gn_args): error(No gn_args found please make sure you have valid configuration.)
     return($$gn_args)
@@ -112,6 +117,7 @@ defineReplace(gnArch) {
     contains(qtArch, "mips"): return(mipsel)
     contains(qtArch, "mips64"): return(mips64el)
     contains(qtArch, "mips64el"): return(mips64el)
+    contains(qtArch, "power64"): return(ppc64)
     return(unknown)
 }
 
@@ -119,6 +125,7 @@ defineReplace(gnOS) {
     macos: return(mac)
     win32: return(win)
     linux: return(linux)
+    freebsd: return(freebsd)
     error(Unsupported platform)
     return(unknown)
 }
